### PR TITLE
WIP: Port CircleCI to v2

### PR DESCRIPTION
**Pull request by stefanv**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      - run:
+          name: test
+          commands: |
+            python -u runtests.py -g --shell -- -c 'make -C doc
+          PYTHON=python html-scipyorg'
+
+# Upload build output to scipy/devdocs repository, using SSH deploy keys.
+# The keys are only available for builds on master branch.
+# https://developer.github.com/guides/managing-deploy-keys/
+# https://circleci.com/docs/1.0/configuration/#ssh-keys
```
Comment (andyfaff) [inline .circleci/config.yml]: docs/1.0 --> docs/2.0?

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+          pip install -q numpy
+          pip install -q nose mpmath argparse Pillow codecov matplotlib Sphinx==1.5.5
+          git submodule init
+          git submodule update
+
+      - run:
+          name: test
+          commands: |
+            python -u runtests.py -g --shell -- -c 'make -C doc
+          PYTHON=python html-scipyorg'
```
Comment (andyfaff) [inline .circleci/config.yml]: Is there a reason to linebreak here?

**Comment by andyfaff** [comment]
Thanks @stefanv. Is the only task of circle CI to build the docs? Does it do any testing?
Did you run the config file through the yaml checker that (I am assuming) they provide?

**Comment by andyfaff** [comment]
Apparently there's a problem with the config file.

**Comment by stefanv** [comment]
Hi, Andrew, I'm still working on this, but will let you know once I think 
it's ready for review.

**Comment by andyfaff** [comment]
What are `shell` and `extra_argv`? at this point? Could be that it can't
find shell or something in extra_argv.
Once you know what both are you can do `which shell`, or PWD to see if it's
can't find out if it actually can't find the shell,or if there's something
in extra_argv which it can't find.

**Comment by pv** [comment]
@stefanv:
```
export SHELL=`which bash`
python -u runtests.py ...
```
or something along those lines.

**Comment by stefanv** [comment]
@andyfaff @pv So far so good; we have a working build up to deployment time.  Not sure how to test that step, though!

**Comment by andyfaff** [comment]
No idea either. You might get a clue from looking at the setup/artefacts
from circle v1.

**Comment by larsoner** [comment]
I set up CircleCI for SciPy originally to build the docs so that we could view them to make sure things look okay with each PR (instead of having to build them locally or wait to see errors once docs were built for release).

As the first necessary additional feature are the artifacts, accessible via the "Artifacts" tab. If you go to the [CircleCI build page](https://circleci.com/gh/scipy/scipy/5843#artifacts/containers/0) for [this PR](https://github.com/scipy/scipy/pull/8481), sign in (if necessary), and click "Artifacts" (if necessary), you'll be able to browse the documentation tree.

Once that works, the next thing needed is the `deployment` step for the `master` branch. This updates the docs on `github` with each merged PR.

I haven't done any 2.0 migrations for my repos yet, though, so I'm not sure how to set these up for 2.0. Hopefully it's not too bad.

**Comment by pv** [comment]
@stefanv: if I read the config correctly, this won't store the artefacts 
for PRs, because the deploy step is limited to branch:master?

The limitation to master branch should only apply to the part that 
uploads the files to scipy/devdocs. The artefacts should be stored also 
for PRs, because then the documentation can be viewed also for PRs.

**Comment by stefanv** [comment]
OK, I moved the artifact storage to the build step.

**Comment by stefanv** [comment]
@pv @andyfaff Can either of you see the artifacts?  That tab does not appear for me, so I wonder if I have the correct permissions.

**Comment by tylerjereddy** [comment]
@stefanv You probably just need to login on the top right corner. It seems to periodically log me out. Permissions should likely be ok as "owner".

Here's a screencap after logging in & clicking the tab for this PR:

image

**Comment by stefanv** [comment]
Logging out and then in again did the trick; thanks @tylerjereddy!

**Comment by pv** [comment]
Thanks, merged. I guess there's no way to test the deploy step before merging.

**Comment by stefanv** [comment]
@pv That's useful to know, thanks; I'll do a PR soon.

**Comment by pv** [comment]
Seems to be broken again, I guess aftermath of the circleci outage.

Was the fix to remove the deploy key from the scipy main repository? I guess it got re-added.

**Pull request merged**